### PR TITLE
[Merged by Bors] - fix(widget): docstring demo doesn't work

### LINKED ISSUE #512: Example in widget docs doesn't work
The following was [reported on Zulip](https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/widget.20component.20example/near/221104512) by @SnobbyDragon. This example from https://leanprover-community.github.io/mathlib_docs/init/meta/widget/basic.html doesn't work:
```lean
meta def Hello : component string empty := component.pure (λ s, ["hello, ", s, ", good day!"])
#html Hello "lean" -- renders "hello, lean, good day!"
```

```
maximum class-instance resolution depth has been reached
(the limit can be increased by setting option 'class.instance_max_depth')
(the class-instance resolution trace can be visualized by setting option 'trace.class_instances')
```

@EdAyers commented:

> So the problem is that when you write Hello "lean", there is a function coercion, but the typeclass inference system is getting stuck in a loop for some reason. I suggest that you just use the clunkier version with html.of_component for now. Maybe someone who actually understands typeclasses knows what needs to change to get this to work.

### PR DESCRIPTION
**Pull request by EdAyers**
fixes #512

Adds a new coercion `component π α → (π → html α)`.

### CODE REVIEW

File: library/init/meta/widget/basic.lean [hunk 1]
```diff
@@ -241,8 +241,13 @@ meta def ignore_action : component π α → component π β
...
 
-meta instance : has_coe (component π empty) (component π α)
-:= ⟨component.filter_map_action (λ p x, none)⟩
+meta instance : has_coe (component π empty) (component π α) :=
+⟨component.filter_map_action (λ p x, none)⟩
+
+meta instance : has_coe_to_fun (component π α) :=
+{ F := λ c, π → html α
+, coe := λ c p, html.of_component p c
+}
```
Comment (gebner) [inline library/init/meta/widget/basic.lean:250]:
```suggestion
{ F := λ c, π → html α,
  coe := λ c p, html.of_component p c }
```

**Review by gebner — COMMENTED** [review]
Why is this a draft?

**Comment by EdAyers** [comment]
I wanted to double check that it doesn't break mathlib but I've done that now

**Pull request closed without merging**